spec.html

```diff
@@ -10942,7 +10965,9 @@ <h1>Static Semantics: StringValue</h1>
             IdentifierName IdentifierPart
         </emu-grammar>
         <emu-alg>
-          1. Return the String value consisting of the sequence of code units corresponding to |IdentifierName|. In determining the sequence any occurrences of `\\` |UnicodeEscapeSequence| are first replaced with the code point represented by the |UnicodeEscapeSequence| and then the code points of the entire |IdentifierName| are converted to code units by UTF16Encoding each code point.
+          1. Let _idText_ be the source text matched by |IdentifierName|.
+          1. Let _idTextUnescaped_ be the result of replacing any occurrences of `\\` |UnicodeEscapeSequence| in _idText_ with the code point represented by the |UnicodeEscapeSequence|.
```

Comment:
> the code point represented by the |UnicodeEscapeSequence|

This language is a little less precise than I would prefer. Could we define a syntax-directed operation like

```
The CPV of UnicodeEscapeSequence :: u Hex4Digits is the code point whose value is (0x1000ℝ times the MV of the first HexDigit) plus (0x100ℝ times the MV of the second HexDigit) plus (0x10ℝ times the MV of the third HexDigit) plus the MV of the fourth HexDigit.
The CPV of UnicodeEscapeSequence :: u { CodePoint } is the code point whose value is the MV of CodePoint.
```

We could also extract that Hex4Digits bit into MV so we don't do the same maths in two places.

Reply:
I like the suggestion, and I'm willing to work on it (have already started), but it's independent of this PR, and it seems to me that doing it properly will be a larger change. So I think it should be in a separate PR.

Reply:
Okay I'm fine with it being done in a separate PR since this PR isn't making anything worse than it already is.

Reply:
Thanks. (BTW, in my preliminary work, I discovered that the spec was already assuming the existence of `MV for Hex4Digits`, thus [this addition](https://github.com/tc39/ecma262/pull/1301/commits/16362aff3698d136637189ab66f70e702dbfe872) to PR #1301.)